internal/argocd/revision.go

```diff
@@ -10,68 +11,163 @@ import (
 	kargoapi "github.com/akuity/kargo/api/v1alpha1"
 	argocd "github.com/akuity/kargo/internal/controller/argocd/api/v1alpha1"
 	"github.com/akuity/kargo/internal/controller/freight"
+	"github.com/akuity/kargo/internal/logging"
 )
 
 // GetDesiredRevision returns the desired revision for the given
 // v1alpha1.Application. If that cannot be determined, an empty string is
 // returned.
-func GetDesiredRevision(
+func GetDesiredRevisions(
 	ctx context.Context,
 	cl client.Client,
 	stage *kargoapi.Stage,
 	update *kargoapi.ArgoCDAppUpdate,
 	app *argocd.Application,
 	frght []kargoapi.FreightReference,
-) (string, error) {
+) ([]string, error) {
+
+	logger := logging.LoggerFromContext(ctx)
+	if app == nil {
+		err := errors.New("Application is nil")
+		logger.Error(err, "Cannot determine desired revision for application, bailing.")
+		return nil, nil
+	}
+
+	appLogger := logger.WithValues("application", app.Name, "namespace", app.Namespace)
```

Comment:
Structured logging only makes sense if the same field names are used consistently. As this field is called `app` elsewhere, it should either be `app` here or `application` in other places.